packages/migration_tool/controllers/single_page/dashboard/system/migration/import.php

```diff
@@ -228,8 +229,17 @@ public function add_content_to_batch()
 
         $importer = \Core::make('migration/manager/importer/parser')->driver($this->request->request('format'));
 
-        if (!is_uploaded_file($_FILES['file']['tmp_name'])) {
-            $this->error->add(t('Invalid XML file.'));
+        $UploadedFile = new UploadedFile(
```

Comment:
Why not
```php
        $uploadedFile = $this->request->files->get('file');
        if ($uploadedFile === null) {
            $this->error->add(t('File not uploaded'));
        } elseif (!$uploadedFile->isValid()) {
            $this->error->add($uploadedFile->getErrorMessage());
        } else {
            $importer->validateUploadedFile($_FILES['file'], $this->error);
        }
```

Reply:
Exactly! Why I did it in a roundabout way:)
Switched to your approach.